File: src/test/fixtures/sample-code.py
# Sample Python code for testing
from typing import List, Optional


class SampleClass:
    """A sample class for testing purposes."""

    def __init__(self, initial_value: int = 0):
        self._value = initial_value

    @property
    def value(self) -> int:
        """Get the current value."""
        return self._value

    @value.setter
    def value(self, new_value: int) -> None:
        """Set a new value."""
        if new_value < 0:
            raise ValueError("Value cannot be negative")
        self._value = new_value

    def increment(self) -> int:
        """Increment the value by 1."""
        self._value += 1
        return self._value

    def decrement(self) -> int:
        """Decrement the value by 1."""
        self._value -= 1
        return self._value

    def reset(self) -> None:
        """Reset the value to 0."""
        self._value = 0


def calculate_sum(numbers: List[int]) -> int:
    """Calculate the sum of a list of numbers."""
    return sum(numbers)


def find_max(numbers: List[int]) -> int:
    """Find the maximum value in a list."""
    if not numbers:
        raise ValueError("List cannot be empty")
    return max(numbers)


def find_min(numbers: List[int]) -> int:
    """Find the minimum value in a list."""
    if not numbers:
        raise ValueError("List cannot be empty")
    return min(numbers)


# Function with potential issues for testing linting
def problematic_function(input_data):
    """Function with linting issues for testing."""
    result = input_data  # Missing type hints
    if result == None:  # Should use 'is None'
        return None
    return result


def unused_function():
    """This function is never used - should be caught by linting."""
    pass

File: archive/v0.2-codebase/src/test/fixtures/sample-code.ts
// Sample TypeScript code for testing
export class SampleClass {
    private value: number;

    constructor(initialValue: number = 0) {
        this.value = initialValue;
    }

    public getValue(): number {
        return this.value;
    }

    public setValue(newValue: number): void {
        if (newValue < 0) {
            throw new Error('Value cannot be negative');
        }
        this.value = newValue;
    }

    public increment(): number {
        this.value++;
        return this.value;
    }

    public decrement(): number {
        this.value--;
        return this.value;
    }

    public reset(): void {
        this.value = 0;
    }
}

export function calculateSum(numbers: number[]): number {
    return numbers.reduce((sum, num) => sum + num, 0);
}

export function findMax(numbers: number[]): number {
    if (numbers.length === 0) {
        throw new Error('Array cannot be empty');
    }
    return Math.max(...numbers);
}

// Function with potential issues for testing linting
export function problematicFunction(input: any): any {
    var result = input; // Should use let/const
    if (result === null) { // Should use ===
        return undefined;
    }
    return result;
}